docs/concepts/policy/pod-security-policy.md

```diff
@@ -1,97 +1,371 @@
+went to create the pod it was not authorized to use the example
+podsecuritypolicy.
+
+In order to fix this, bind the `psp:unprivileged` role to the pod's service
+account instead. In this case (since we didn't specify it) the service account
+is `default`:
+
+```shell
+$ kubectl-admin create rolebinding default:psp:unprivileged \
+    --role=psp:unprivileged \
+    --serviceaccount=psp-example:default
+rolebinding "default:psp:unprivileged" created
+```
+
+Now if you give it a minute to retry, the replicaset-controller should
+eventually succeed in creating the pod:
+
+```shell
+$ kubectl-user get pods --watch
+NAME                    READY     STATUS    RESTARTS   AGE
+pause-7774d79b5-qrgcb   0/1       Pending   0         1s
+pause-7774d79b5-qrgcb   0/1       Pending   0         1s
+pause-7774d79b5-qrgcb   0/1       ContainerCreating   0         1s
+pause-7774d79b5-qrgcb   1/1       Running   0         2s
+^C
+```
+
+### Clean up
+
+Delete the namespace to clean up most of the example resources:
+
+```shell
+$ kubectl-admin delete ns psp-example
+namespace "psp-example" deleted
+```
+
+Note that `PodSecurityPolicy` resources are not namespaced, and must be cleaned
+up separately:
+
+```shell
+$ kubectl-admin delete psp example.unprivileged
+podsecuritypolicy "example.unprivileged" deleted
+```
+
+### Example Policies
+
+This is the least restricted policy you can create, equivalent to not using the
+pod security policy admission controller:
+
+{% include code.html language="yaml" file="privileged-psp.yaml" ghlink="/docs/concepts/policy/privileged-psp.yaml" %}
+
+This is an example of a restrictive policy that requires users to run as an
+unprivileged user, blocks possible escalations to root, and requires use of
+several security mechanisms.
+
+{% include code.html language="yaml" file="restricted-psp.yaml" ghlink="/docs/concepts/policy/restricted-psp.yaml" %}
+
+## Policy Reference
+
+### Host namespaces
+
+**HostPID** - Controls whether the pod containers can share the host process ID
+namespace. Note that when paired with ptrace this can be used to escalate
+privileges outside of the container.
+
+**HostIPC** - Controls whether the pod containers can share the host IPC
+namespace.
+
+**HostNetwork** - Controls whether the pod may use the node network
+namespace. Doing so gives the pod access to the loopback device, services
+listening on localhost, and could be used to snoop on network activity of other
+pods on the same node.
+
+**HostPorts** - Provides a whitelist of ranges of allowable ports in the host
+network namespace. Defined as a list of `HostPortRange`, with `min`(inclusive)
+and `max`(inclusive). Defaults to no allowed host ports.
+
+**AllowedHostPaths** - See [Volumes and file systems](#volumes-and-file-systems).
+
+### Volumes and file systems
+
+**Volumes** - Provides a whitelist of allowed volume types. The allowable values
+correspond to the volume sources that are defined when creating a volume:
```

Comment:
This list is already outdated :-(

I think that it would be better to put a link to another page: https://kubernetes.io/docs/concepts/storage/volumes/#types-of-volumes In this case, we make this page shorter, less outdated, and reduce the number of places where a new volume types should be updated.

Maybe we can list some most frequently used types, mention `*` and that's would be enough.

Reply:
Good call, done.